### PR TITLE
Add `ycsb` runner and add to CI

### LINKED ISSUE #425: Run go-ycsb
https://github.com/pingcap/go-ycsb

Do the most basic simplest fastest integration into the dance tool (including a new runner) and CI. We will tweak and change everything later, but we need to start somewhere.

### PR DESCRIPTION
**Pull request by b1ron**
Closes #425.

### CODE REVIEW

**Comment by b1ron** [comment]
We can ignore [this](https://github.com/FerretDB/dance/pull/449#issuecomment-1562995644) comment as all required tests are now passing however there's still a flaky test which #442 aims to resolve.

File: cmd/dance/main.go [hunk 1]
```diff
@@ -131,6 +132,8 @@ func main() {
 			runRes, err = jstest.Run(ctx, dir, config.Args)
 		case "service":
 			runRes, err = service.Run(ctx, dir, config.Args)
+		case "ycsb":
```
Comment (AlekSi) [inline cmd/dance/main.go:135]: That should be a constant now.

Please resolve merge conflicts

File: internal/ycsb/ycsb.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+		Output: "",
+	}
+
+	return res, nil
+}
+
+// runWorkload loads and runs a YCSB workload. Properties defined in the YAML file
+// will override properties defined in the workload parameter file.
+func runWorkload(dir string, args ...string) ([]byte, error) {
+	bin, err := exec.LookPath("go-ycsb")
```
Comment (AlekSi) [inline internal/ycsb/ycsb.go:51]: We shouldn't lookup path, but use a fixed relative path instead

File: internal/ycsb/ycsb.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+// Run runs YCSB workloads.
+// It loads and runs a YCSB workload. Properties defined in the YAML file
+// will override properties defined in the workload parameter file.
+func Run(ctx context.Context, dir string, args []string) (*internal.TestResults, error) {
+	res := &internal.TestResults{
+		TestResults: make(map[string]internal.TestResult),
+	}
+
+	_, err := os.Stat("../bin/go-ycsb")
```
Comment (AlekSi) [inline internal/ycsb/ycsb.go:36]: Let's use `filepath.Join` everywhere to make it work on Windows

File: internal/ycsb/ycsb.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	wlArgs := []string{"load", "mongodb", "-P", wlFile}
+	wlArgs = append(wlArgs, "-p")
+	wlArgs = append(wlArgs, args[1:]...)
+
+	cmd := exec.CommandContext(ctx, bin, wlArgs...)
+	cmd.Dir = dir
+
+	log.Printf("Loading workload with properties %s", strings.Join(args, " "))
+
+	// load the workload
```
Comment (AlekSi) [inline internal/ycsb/ycsb.go:54]: I think that comment should be above to match the position of the pair comment below

File: internal/ycsb/ycsb.go [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+	err = cmd.Run()
+	if err != nil {
```
Comment (AlekSi) [inline internal/ycsb/ycsb.go:56]:
```suggestion
	if err = cmd.Run(); err != nil {
```

File: internal/ycsb/ycsb.go [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+	cmd.Dir = dir
+
+	log.Printf("Running %s", strings.Join(cmd.Args, " "))
+
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return nil, err
+	}
+
+	log.Print(string(out))
```
Comment (AlekSi) [inline internal/ycsb/ycsb.go:72]: Why do we print it instead of returning?

File: internal/ycsb/ycsb.go [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+	// the run phase will execute the workload and will report performance statistics on stdout
+	cmd.Args[1] = "run"
+	cmd = exec.CommandContext(ctx, bin, cmd.Args[1:]...)
+	cmd.Dir = dir
+
+	log.Printf("Running %s", strings.Join(cmd.Args, " "))
+
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return nil, err
```
Comment (AlekSi) [inline internal/ycsb/ycsb.go:69]: Why don't we return a non-zero exit code as a Fail result as we do in the command runner?

File: tests/ycsb.yml
Comment (AlekSi) [inline tests/ycsb.yml:1]: Let's create a couple of files with different workloads from the very beginning as an example

File: tests/ycsb/workloads/minio
Comment (AlekSi) [inline tests/ycsb/workloads/minio:1]: I think we don't have a need to include workloads we are never going to use

File: internal/ycsb/ycsb.go (hunk 3 above)
Comment (b1ron) [reply internal/ycsb/ycsb.go:36]: Oh yes, complete oversight on my part.

File: internal/ycsb/ycsb.go (hunk 7 above)
Comment (b1ron) [reply internal/ycsb/ycsb.go:69]: I suppose we could do that for consistency

File: tests/ycsb.yml
Comment (b1ron) [reply tests/ycsb.yml:1]: Please see [ycsb1.yml](https://github.com/FerretDB/dance/pull/449/files#diff-fca50f7884bcff5c7daa134cb9a05873dad2d5972ad1365b46f4136dba41e093) and [ycsb2.yml](https://github.com/FerretDB/dance/pull/449/files#diff-6225c7120984b583b16496b30963b7eeb5dd67a1a338f9f9a75f7197d0a6f3b4). First once is update heavy workload, second is read only workload.

File: internal/ycsb/ycsb.go (hunk 6 above)
Comment (b1ron) [reply internal/ycsb/ycsb.go]: I removed that

File: internal/ycsb/ycsb.go (hunk 4 above)
Comment (b1ron) [reply internal/ycsb/ycsb.go]: Fixed

File: internal/ycsb/ycsb.go (hunk 5 above)
Comment (b1ron) [reply internal/ycsb/ycsb.go]: Fixed

File: .github/workflows/dance.yml [hunk 8]
```diff
@@ -41,6 +41,7 @@ jobs:
           - mongo
           - mongo-go-driver
           - mongo-tools
+          - ycsb
```
Comment (AlekSi) [inline .github/workflows/dance.yml:44]:
```suggestion
          - ycsb1
          - ycsb2
```

File: internal/ycsb/ycsb.go [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+// Run runs YCSB workloads.
+// It loads and runs a YCSB workload. Properties defined in the YAML file
+// will override properties defined in the workload parameter file.
+func Run(ctx context.Context, dir string, args []string) (*internal.TestResults, error) {
+	res := &internal.TestResults{
+		TestResults: make(map[string]internal.TestResult),
+	}
+
+	bin := filepath.Join("../", "/bin/go-ycsb")
```
Comment (AlekSi) [inline internal/ycsb/ycsb.go:38]:
```suggestion
	bin := filepath.Join("..", "bin", "go-ycsb")
```

File: internal/ycsb/ycsb.go [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+	bin := filepath.Join("..", "bin", "go-ycsb")
+
+	_, err := os.Stat(bin)
+	if err != nil {
+		return nil, err
+	}
+
+	// because we set cmd.Dir, the relative path here is different
+	bin = filepath.Join("../", bin)
```
Comment (AlekSi) [inline internal/ycsb/ycsb.go:46]:
```suggestion
	bin = filepath.Join("..", bin)
```

https://pkg.go.dev/path/filepath#Join

**Review by AlekSi — APPROVED** [review]

**Pull request merged**